components/hooks/useStoryReader.ts

```diff
@@ -69,17 +87,31 @@ export const useStoryReader = (id: string) => {
       setSeekValue(duration ? (actualTime / duration) * 100 : 0);
     };
 
-    const handleAudioEnd = () => setIsPlaying(false);
+    const handleAudioEnd = () => {
+      // console.log('The end');
+      setIsPlaying(false);
+      setAudioEnded(true);
+      setIsFirstEnd(false);
+    };
 
     audio.current = new Audio(source);
     audio.current.ontimeupdate = handleTimeUpdate;
     audio.current.onended = handleAudioEnd;
-
     return () => {
       stopStory();
     };
   }, [source, stopStory]);
 
+  useEffect(() => {
```

Comment:
I tested the App a bit. When you finished the story then try to move slider back. Then you get Story-Finished event hundred times. Replace temporarily plausible function with this `const plausible = (event,data) => console.log(event);` and test it . Useffects are trickiest part of react. I would start to look there. When you need to set variable included in condition in Useffect inside same Useffects use const switch = useRef(true); . This variable doesn have to be included in depc so it cant make infinite loop. Try to look at my tracking hook /components/basecomponents/Exercise/utils/useTracking.ts

Reply:
okay, I`ll think about this.
thanks

Reply:
Good job.